### PR TITLE
Fully async

### PR DESCRIPTION
**Pull request by slang25**
This PR makes JustBehave no longer block on async code, this should allow consumers (JustSaying for example) to have non-blocking startup code, for teardown you'll need to rename `PostAssertTeardown` -> `PostAssertTeardownAsync` and return a `Task` when using the async base class.

This also adds strong naming. My hope is for this to be a good candidate for release (after a beta), fixing the outstanding major issues.

### CODE REVIEW

File: src/JustBehave/AsyncBehaviourTestBase.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ protected virtual Layout LogLayout()
             return "${message}";
         }
 
-        protected virtual void PostAssertTeardown() { }
+        protected virtual Task PostAssertTeardownAsync() => Task.FromResult(true);
```
Comment (martincostello) [inline src/JustBehave/AsyncBehaviourTestBase.cs]: `Task.CompletedTask`? Or is that not supported by all the TFMs?

Comment (slang25) [reply src/JustBehave/AsyncBehaviourTestBase.cs]: Not supported by the net452 tfm, might create an internal static

**Comment by martincostello** [comment]
Before I forget, before publishing this repo will need a major version bump as adding strong naming is a breaking change.

**Comment by slang25** [comment]
So at the moment, the last version was 2.0.0-beta-51, the previous commit drops the beta and makes in the final release, but we never pushed it to nuget. So I could add the beta back now, just in case there are any changes needed before making it 2.0.0. While we are in beta we are still ok to make braking changes according with SemVer.

**Pull request merged**